tensorboard/webapp/runs/views/runs_selector/runs_selector_container.ts

```diff
@@ -49,6 +71,51 @@ export class RunsSelectorContainer {
       ].filter((col) => col !== null) as RunsTableColumn[];
     })
   );
+  private readonly ngUnsubscribe = new Subject();
+
+  constructor(
+    private readonly store: Store<State>,
+    private readonly snackBar: MatSnackBar
+  ) {}
+
+  ngOnInit() {
+    // Notify the user that new runs may not be selected. Avoid showing it too
+    // often, since it would be annoying to see the alert re-appear on every
+    // auto-reload (assuming a new run per reload).
+    const runsExceedsLimitForRoute$ = this.experimentIds$.pipe(
+      takeUntil(this.ngUnsubscribe),
+      switchMap(() => {
+        // Returns an Observable that emits once and completes when the current
+        // route's run count goes over the limit.
+        return this.store.select(getCurrentRouteRunSelection).pipe(
+          distinctUntilChanged((a, b) => {
+            return a === b;
+          }),
+          filter((runSelectionMap: Map<string, boolean> | null) => {
+            if (!runSelectionMap) {
+              return false;
+            }
+            return runSelectionMap.size > MAX_NUM_RUNS_TO_ENABLE_BY_DEFAULT;
+          }),
+          take(1)
+        );
+      })
+    );
+    runsExceedsLimitForRoute$.subscribe(() => {
+      const text =
+        `The number of runs is over ` +
+        `${MAX_NUM_RUNS_TO_ENABLE_BY_DEFAULT}. New runs are unselected ` +
+        `for performance reasons.`;
+      this.snackBar.open(text, 'DISMISS', {
+        duration: 5000,
+        horizontalPosition: 'start',
+        verticalPosition: 'bottom',
+      });
```

Comment:
Can we undo this and use the alerts module? This is taking the imperative approach of a view instead of more declarative version: e.g., you are using business logic to determine whether to show a snackbar or not then are making an imperative API call to the snackbar. Instead, the view should react to the data change which we already have in the alerts component. Just add an alert in this case?

Reply:
Done.